packages/core/upload/server/services/provider.js

```diff
@@ -5,7 +5,7 @@ const { streamToBuffer } = require('../utils/file');
 
 module.exports = ({ strapi }) => ({
   async upload(file) {
-    if (isFunction(strapi.plugin('upload').provider.uploadStream)) {
+    if (isFunction(strapi.plugin('upload').provider.original.uploadStream)) {
```

Comment:
ah, interesting way to handle the optional uploadStream! Thanks for tackling this.